nemo/collections/asr/parts/submodules/transformer_modules.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import torch
+from torch import nn as nn
+from torch.nn import LayerNorm
+
+from nemo.collections.asr.parts.submodules.conformer_modules import ConformerFeedForward
+from nemo.collections.asr.parts.submodules.multi_head_attention import (
+    MultiHeadAttention,
+    RelPositionMultiHeadAttention,
+)
+from nemo.collections.asr.parts.utils.activations import Swish
+
+__all__ = ['TransformerEncoderLayer']
+
+
+class TransformerEncoderLayer(torch.nn.Module):
```

Comment:
@stevehuang52 Could you check if this transformer module is the same that you are using for SLU?

Reply:
> @stevehuang52 Could you check if this transformer module is the same that you are using for SLU?

@jbalam-nv This one is  mostly the same as the one in `nemo.collections.nlp.modules.common.transformer.transformer_encoder.TransformerEncoder`, but adds a FFN before the transformer. I also use TransformerDecoder and SequenceGenerator from `nemo.collections.nlp.modules.common.transformer` in SLU. I think we can move the whole transformer module to `nemo.collections.common` and avoid reimplementing similar stuffs.

Reply:
@krishnacpuvvada Steve has a PR to move transformer from nlp to common, could you please check if you could reuse some of that implementation?